tests/BUILD.bazel

```diff
@@ -389,3 +389,49 @@ java_fuzz_target_test(
     allowed_findings = ["com.code_intelligence.jazzer.api.FuzzerSecurityIssueHigh"],
     target_class = "com.example.SilencedFuzzer",
 )
+
+java_library(
+    name = "OfflineInstrumentedTarget",
+    srcs = ["src/test/java/com/example/OfflineInstrumentedTarget.java"],
+)
+
+genrule(
+    name = "OfflineInstrumentedTargetInstrumented",
+    srcs = [
+        ":libOfflineInstrumentedTarget.jar",
+    ],
+    outs = ["OfflineInstrumentedTargetInstrumented.jar"],
+    cmd = """
+set -euo pipefail
+
+export JAVA_HOME=$$(realpath $(JAVABASE))
+export PATH="$$JAVA_HOME/bin:$$PATH"
+
+jacococli=$$(realpath external/jacococli/file/jacococli.jar)
+java -jar $$jacococli instrument "$<" --dest "$$PWD"
+
+# Copy instrumented jar to expected output destination
+cp $$PWD/$$(basename "$<") $@
+""",
+    target_compatible_with = LINUX_ONLY,
+    toolchains = [
+        "@bazel_tools//tools/jdk:current_java_runtime",
+    ],
+    tools = [
+        "@jacococli//file:jacococli.jar",
```

Comment:
If you wrap this jar in a `java_binary` target and add that to `tools`, the genrule should become much simpler as you won't need to get and use the  Java toolchain anymore.

Reply:
I'm not sure how to do that. `java_binary` requires `srcs` or `main_class` which we don't have here, or do we? Simplifying the `genrule` would be great, though.

Reply:
I pushed a commit that implements this, please take a look and feel free to make changes.

Reply:
Much better, thx! With your change the script also works on all platforms.